jobfunnel/tests/test_validate.py

```diff
@@ -20,12 +21,180 @@
     [['search_terms', 'region', 'province'], None],
     [['delay_config', 'function'], 'weird'],
     [['delay_config', 'min_delay'], 50.0],
+    [['delay_config', 'min_delay'], -1],
+    [['delay_config', 'delay'], 2],
+    [['max_listing_days'], -1],
+    [['data_path'], 'data_dump'],
+    [['duplicate_list_path'], 'duplicate_list_.csv'],
+    [['log_path'], 'data/jobfunnel_.log'],
+    [['filter_list_path'], 'data/filter_list_.json']
 ]
 
 configs = config_factory(config, attr_list)
 
+# Test all paths with invalid values
+
```

Comment:
There seems to be excessive whitespace after a comment line. I like how you divide the tests into groups with a corresponding comment but I would reduce the number of blank lines after a comment to 1 and two blank lines before a comment.

Reply:
Thanks so much for your awesome feedback!
Just fixed the issue. But I have a question before I push:should I bump up the version number after this change?
The reason why I'm not sure is because we haven't completed/fixed the entire issue of test coverage, just a part of it. I lean to think to not update the version number until we fix the entire issue of test coverage as a whole, but I could be wrong about this.

Reply:
since this isn't a feature, more something that should already be in the codebase (i.e. tech-debt), IMO we shouldn't bump the version for this

Reply:
Yes we do not have to update the version number. I believe the version number system is typically something like <MAJOR>.<MINOR>.<BUGS> where a major version updates indicates a major API design change, a minor change is adding functionalities (without breaking the older ones) and bugs should merely fix issues in the code. 

Testing is therefore a bit of a weird case since it does none of these things. It is merely for internal testing for the developers to check if everything is still alright. But of course you could argue that testing is a type of bug fixing but I agree that updating the version is unnecessary in this case.

One more thing: while I did mention all functions in the increase test coverage issue #55 it is generally not necessary to test every single line of your code. It is simply too time expensive for little return. (Unless you're designing airplanes haha)

Reply:
:+1: